poleno/attachments/models.py

```diff
@@ -109,6 +111,8 @@ def save(self, *args, **kwargs):
                 self.created = utc_now()
             self.size = self.file.size
             self.content_type = magic.from_buffer(self.file.read(), mime=True)
+            if mimetypes.guess_type(self.name)[0] != self.content_type:
+                self.name = os.path.splitext(self.name)[0] + guess_extension(self.content_type, default=".bin")
```

Comment:
Na druhe zamyslenie: Mali by sme sanitizovat cely `name`, nie len prilohu. Preto pls vygoogli ake znaky su v nazvoch suborov zakazane a vyhod ich z `name`. Resp. pozri, ci na to nie je nejaka standardna funkcia. Hlavne ide o znaky ako `\n`, `\0`, ale aj dalsie divne znaky. Pozor, aby prezili bezne unicode znaky. Rovnaku sanitizaciu sprav aj v migracii.

Reply:
`name` teda sanitizujme takto:
- Vyhodme vsetky znaky s `ord(c) < 32`. Cize vsetky kontrolne znaky. Ale medzeru (32) uz nechavame.
- Ak je `name` bez pripony dlhsie ako 100 znakov, skratime ho (zprava) na 100 znakov a vratime priponu.

Reply:
zkratenie zprava znamena, ze retazec 'abcdefgh' skrateny zprava na 3 znaky mi vrati 'fgh' ?

Reply:
@viliambalaz Spresnenie: Spracenie sprava znamena, ze z pravej strany znaky vyhadzuejem `abcdef` -> `abc`